helpers/stream.go

```diff
@@ -0,0 +1,56 @@
+package helpers
+
+import (
+	"errors"
+	"io"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+// EOFTimeout is the maximum amount of time to wait to successfully observe an
+// EOF on the stream. Defaults to 60 seconds.
+var EOFTimeout = time.Second * 60
```

Comment:
this will break backwards compatibility, the variable needs to be set in `core` and can't be set at the leaf.

Reply:
I recommend removing all such configuration variables from the leaves.
Better to break the build than have an assignment that has no effect.

Reply:
We can make them constants.

Reply:
Can we make them constants in the leaves while being initialized by a variable?
If that's the case, then it would be fine as it would break assignments.
But we really don't want to turn them into constants in the core module, they are variable for good reason.

Reply:
Hm. No, we can't. I mean, ideally we'd have some kind of global config...

Reply:
I think we should intentionally break the donor modules by removing those vars at origin. Much better than a silent change in behaviour.